frame/bridge/s2s/relay/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+		type OutboundMessageFee: From<u64>;
+
+		type CallToPayload: CallToPayload<Self::OutboundPayload>;
+		type BridgedAssetReceiverT: BridgedAssetReceiver<RelayAccount<AccountId<Self>>>;
+		type BridgedAccountIdConverter: Convert<H256, Self::AccountId>;
+		type ToEthAddressT: ToEthAddress<Self::AccountId>;
+		type MessageRelayCallT: MessageRelayCall<Self::OutboundPayload, Self::Call>;
+	}
+
+	#[pallet::event]
+	#[pallet::metadata(T::AccountId = "AccountId")]
+	pub enum Event<T: Config<I>, I: 'static = ()> {
+		/// new message relayed
+		NewMessageRelayed(T::AccountId, u8),
+	}
+
+	#[pallet::error]
+	pub enum Error<T, I = ()> {
+		/// The proof is not in backing list
+		InvalidProof,
+		/// Invalid Backing address
+		InvalidBackingAddr,
+		/// Encode Invalid
+		EncodeInvalid,
+		/// Dispatch Message Relay Failed
+		DispatchFailed,
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn remote_root_id)]
+	pub type RemoteRootId<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, T::AccountId, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config<I>, I: 'static = ()> {
+		pub phantom: PhantomData<(T, I)>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config<I>, I: 'static> Default for GenesisConfig<T, I> {
+		fn default() -> Self {
+			Self {
+				phantom: Default::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I> {
+		fn build(&self) {
+			let chain_id = T::BridgedChainId::get();
+			let hex_id = derive_account_id::<T::AccountId>(chain_id, SourceAccount::Root);
+			let target_id = T::BridgedAccountIdConverter::convert(hex_id);
+			<RemoteRootId<T, I>>::put(target_id);
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {}
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {}
+}
+
+impl<T, I> Relay for Pallet<T, I>
+where
+	T: Config<I>,
+	I: 'static,
+	<T::Call as Dispatchable>::Origin: From<RawOrigin<T::AccountId>>,
+	T::Call: Dispatchable<PostInfo = PostDispatchInfo>,
+{
+	type RelayOrigin = AccountId<T>;
+	type RelayMessage = (u32, Token, RelayAccount<AccountId<T>>);
+
+	fn verify_origin(proof: &Self::RelayOrigin) -> Result<EthereumAddress, DispatchError> {
+		let source_root = <RemoteRootId<T, I>>::get();
+		ensure!(&source_root == proof, <Error<T, I>>::InvalidProof);
+		Ok(T::ToEthAddressT::into_ethereum_id(proof))
+	}
+
+	fn relay_message(message: &Self::RelayMessage) -> Result<(), DispatchError> {
+		let (spec_version, token, relay_account) = message.clone();
+		let encoded = T::BridgedAssetReceiverT::encode_call(token, relay_account)
+			.map_err(|_| <Error<T, I>>::EncodeInvalid)?;
+		let relay_id: AccountId<T> = T::PalletId::get().into_account();
+		let payload = T::CallToPayload::to_payload(spec_version, encoded);
+
+		T::MessageRelayCallT::encode_call(payload)
+			.dispatch(RawOrigin::Signed(relay_id).into())
+			.map_err(|_| <Error<T, I>>::DispatchFailed)?;
```

Comment:
Is here going to do remote call or local call?

If it is an remote call, why there is no pallet-bridge-messages's `send_message `?

Dispatch and Message are two different things, Dispatch is to call send from remote, Message is it to send call as message to remote.

This is why I think we do not need this pallet, we should use `pallet-bridge-messages` and `pallet-bridge-dispatch` directly from Backing and Issuing

Reply:
It is local call. Yes there is no trait for pallet-bridge-messages's `send_message`. And we don't want to modify pallet-bridge-messages's code. So we use a dispatch call here.